### DISCUSSION TITLE
Raspberry Pico W program not working after updating to firmware v1.24

### DISCUSSION THREAD

**Discussion by visogonzalez** [micropython/micropython#16202] [RP2040 / Pico]
I have a very simple program for Raspberry Pico W that toogles the state of two GPIO when an UDP message is received through the board WiFi AP. It is not working after updating to the last v1.24 firmware and I really don't know why. I can connect to the Pico AP but it doesn't receive any mesages, It works if I go back to v1.23, so it is clearly something with v1.24. I can't see anything in the firmware documentation that explains this.

Any ideas?

The code:
```
import network
import time
import socket
from machine import Pin

r1=Pin(16,Pin.OUT)
r2=Pin(17,Pin.OUT)
led2= Pin("LED", Pin.OUT)
led2.on()
r1.off()
r2.on()

def ap_mode(ssid, password):

   ap = network.WLAN(network.AP_IF)
   ap.config(essid=ssid, password=password, pm = 0x111022)
   ap.active(True)

   while ap.active() == False:
      pass
   print('AP Mode Is Active, You can Now Connect')
   print('IP Address To Connect to:: ' + ap.ifconfig()[0])

   s2 = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
   s2.bind(('', 5005))

   while True:
   data, addre = s2.recvfrom(1024)
   r1.toggle()
   r2.toggle()

ap_mode('DISPARA','DISPARAFILM')
```

**Comment by scruss** [comment]
please see answer here: [Raspberry Pico W program not working after updating to firmware v1.24 - Raspberry Pi Forums](https://forums.raspberrypi.com/viewtopic.php?p=2267225#p2267225)

Summary (confirmed by the OP to have been successful): turn the station off if you use the AP:
```py3
    sta_if = network.WLAN(network.STA_IF)
    sta_if.active(False)
```

**Comment by bidrohini** [comment]
The pm (power management) configuration in ap.config() might be affecting the network behavior in the new firmware. Try removing or modifying the pm configuration:

ap.config(essid=ssid, password=password)

**Reply by visogonzalez** [reply]
Thanks for your message. The problem with the new firmware was related with bind property of the socket. I changed s.bind((' ', 5005)) to s.bind(('0.0.0.0', 5005)).
